File: challenges/url-shortener/go/starter/exercise.go
package exercise

// Add the state required by the challenge contract.
type URLShortener struct{}

func NewURLShortener() *URLShortener {
	panic("Not implemented")
}

func (s *URLShortener) Shorten(url string) string {
	panic("Not implemented")
}

func (s *URLShortener) Resolve(code string) (string, bool) {
	panic("Not implemented")
}

File: challenges/url-shortener/python/starter/exercise.py
"""Implement the contract in the challenge README. Reference code is separate."""
class URLShortener:

    def __init__(self):
        raise NotImplementedError('Not implemented: __init__')

    def shorten(self, url: str) -> str:
        raise NotImplementedError('Not implemented: shorten')

    def resolve(self, code: str) -> str | None:
        raise NotImplementedError('Not implemented: resolve')
